### ISSUE TITLE
Docstring generation not working if the return type is an Annotated containing a function call

### ISSUE THREAD

**Issue by Richardk2n** [spyder-ide/spyder#14520]
### Issue Report Checklist

* [x] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
* [x] Read the relevant sections of the [Spyder Troubleshooting Guide](https://github.com/spyder-ide/spyder/wiki/Troubleshooting-Guide-and-FAQ) and followed its advice
* [x] Reproduced the issue after updating with ``conda update spyder`` (or ``pip``, if not using Anaconda)
* [ ] Could not reproduce inside ``jupyter qtconsole`` (if console-related)
* [ ] Tried basic troubleshooting (if a bug/error)
    * [x] Restarted Spyder
    * [ ] Reset preferences with ``spyder --reset``
    * [ ] Reinstalled the latest version of [Anaconda](https://www.anaconda.com/download/)
    * [ ] Tried the other applicable steps from the Troubleshooting Guide
* [x] Completed the **Problem Description**, **Steps to Reproduce** and **Version** sections below

## Problem Description

If a function or method return type uses Annotated wit a function call docstring generation fails.
So `def test(self) -> Annotated[str, int("2")]:` fails while `def test(self) -> Annotated[str, "2"]:` works.
Fails in this case means does nothing (the shortcut produces only """ and if you type """ The generate docstring 'button' shows up but does nothing).

### What steps reproduce the problem?

1. type in the examples
2. try to generate a docstring
3. 

### What is the expected output? What do you see instead?

A docstring
No docstring

### Paste Traceback/Error Below (if applicable)

```python-traceback

PASTE TRACEBACK HERE

```

## Versions

* Spyder version: 4.2.1
* Python version: 3.9.1
* Qt version: 5.12.10
* PyQt version: 5.12.3
* Operating System name/version: Windows 10

### Dependencies

```
PASTE DEPENDENCIES HERE
```

**Comment by steff456** [comment]
Hi @Richardk2n,

Can you please provide an example with the expected output?

**Comment by Richardk2n** [comment]
@steff456 In my opinion it should be:
```
def test(self) -> Annotated[str, int("2")]:
    """
    

    Returns
    -------
    Annotated[str, int("2")]
        DESCRIPTION.

    """
```

**Comment by Richardk2n** [comment]
Also apperently what I wrote above is wrong def `test(self) -> Annotated[str, "2"]:` only "works" in so far as it generates a docstring, however the return type in that docsting is labeled as "None". This is as far as i can tell alwas the case, when there is a string literal in the annotated describing the return type.

**Comment by ccordoba12** [comment]
@ok97465, could you take a look at this one? Thanks!

**Comment by ok97465** [comment]
@ccordoba12  Sorry for the late reply. I'm in a very busy situation right now, so I can't make time. If my problem is solved, I will check this problem.

**Comment by ccordoba12** [comment]
@ok97465, don't worry, there's no rush. I just wanted to let you know about this issue.

**Comment by eafpres** [comment]
I think I'm having an issue that is related:
I'm using Google format for docstring
Here you can see with a normal function, it says Returns: None.

```
def func_w_dosctring(var, varb, some_list):
"""
  

  Args:
    var (TYPE): DESCRIPTION.
    varb (TYPE): DESCRIPTION.
    some_list (TYPE): DESCRIPTION.

  Returns:
    None.

  """
#
# a function with a docstring
#
  result = 42
#
  return result
```

**Comment by eafpres** [comment]
For the repro I posted above, I found the issue:
if the opening triple quotes are not indented, it generates the docstring as shown with Returns; None.
if the opening triple quotes are indented, it works correctly.

Not sure if this is intended or valid behavior.

**Comment by CAM-Gerlach** [comment]
Three and a half years late on this, sorry, but I opened a new issue, #25617 for the behavior reported by @eafpres as it is unrelated to the original problem and I can repro it by triggering "triple-quote completion" with the triple quotes at the wrong indent level as shown in his example, including after all the fixes and improvements in #25558 .
